nixos/modules/services/networking/dnsdist.nix

```diff
@@ -35,25 +35,18 @@ in {
     };
   };
 
-  config = mkIf config.services.dnsdist.enable {
-    systemd.services.dnsdist = {
-      description = "dnsdist load balancer";
-      wantedBy = [ "multi-user.target" ];
-      after = ["network.target"];
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.dnsdist ];
 
+    systemd.services.dnsdist = {
       serviceConfig = {
-        Restart="on-failure";
-        RestartSec="1";
+        # compatibility with this module pre 21.03
         DynamicUser = true;
-        StartLimitInterval="0";
-        PrivateDevices=true;
-        AmbientCapabilities="CAP_NET_BIND_SERVICE";
-        CapabilityBoundingSet="CAP_NET_BIND_SERVICE";
-        ExecStart = "${pkgs.dnsdist}/bin/dnsdist --supervised --disable-syslog --config ${configFile}";
-        ProtectHome=true;
-        RestrictAddressFamilies="AF_UNIX AF_INET AF_INET6";
-        LimitNOFILE="16384";
-        TasksMax="8192";
+        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" "CAP_SETGID" "CAP_SETUID" ];
```

Comment:
The upstream [unit](https://github.com/PowerDNS/pdns/blob/master/pdns/dnsdistdist/dnsdist.service.in) doesn't list`CAP_SETGID` and `CAP_SETUID`. Are they needed?

Reply:
They are not.

https://dnsdist.org/running.html
> dnsdist can drop privileges using the --uid and --gid command line switches to ensure it does not run with root privileges. Note that dnsdist drops its privileges after parsing its startup configuration and binding its listening and initial newServer() sockets as user root. It is highly recommended to create a system user and group for dnsdist. Note that most packaged versions of dnsdist already create this user.

Specifically, you can see from [this commit](https://github.com/PowerDNS/pdns/pull/7820/files#diff-6788ff0c0758df99586902c3eb978366L23) in [this PR](https://github.com/PowerDNS/pdns/pull/7820) they are not required anymore.

**NOTE:** I don't run `powerdns` or `dnsdist`, but have been evaluating the possibility of using them and noticed they were out of date in NixOS when compared to upstream (or other distros, for that matter).